web/server/codechecker_server/api/report_server.py

```diff
@@ -2534,20 +2557,44 @@ def getLinesInSourceFileContents(self, lines_in_files_requested, encoding):
         self.__require_view()
         with DBSession(self._Session) as session:
             res = defaultdict(lambda: defaultdict(str))
-            for lines_in_file in lines_in_files_requested:
-                if lines_in_file.fileId is None:
-                    LOG.warning("File content requested without a fileId.")
-                    LOG.warning(lines_in_file)
-                    continue
-                sourcefile = session.query(File).get(lines_in_file.fileId)
-                cont = session.query(FileContent).get(sourcefile.content_hash)
+
+            contents = []
+            for chunk in util.chunks(
+                    lines_in_files_requested, SQLITE_MAX_VARIABLE_NUMBER):
+                contents.extend(
+                    session.query(FileContent.content, File.id)
+                        .join(
+                            File,
+                            FileContent.content_hash == File.content_hash)
+                        .filter(File.id.in_(
+                                [line.fileId if line.fileId is not None
+                                    else LOG.warning(
+                                        f"File content "
+                                        "requested without fileId {l}")
+                                    for line in chunk]))
+                    .all())
+
+            lines_in_files_requested.sort(key=lambda line: line.fileId)
+            contents.sort(key=lambda c: c.id)
+
+            for file in lines_in_files_requested:
+                res[file.fileId]
```

Comment:
You were right, this really inserts the given key with a default constructed value. However, this loop is still unnecessary, because the insertion happens in line 2596 anyways.

Reply:
Removed it,